tket2/src/rewrite_space/factory/kak.rs

```diff
@@ -0,0 +1,255 @@
+//! KAKResynthesis explorer for rewrite space exploration.
+use hugr::{
+    builder::{
+        endo_sig, DFGBuilder, Dataflow, DataflowHugr
+    }, 
+    extension::{prelude::qb_t, simple_op::MakeExtensionOp},
+    Hugr, HugrView, IncomingPort, OutgoingPort
+};
+use hugr::Direction;
+use tket2::{
+    portmatching::PatternMatch, rewrite_space::{
+        CommitFactory, IterMatchedWires, PersistentHugr, Walker
+    }, Tk2Op
+};
+use hugr::persistent::{PersistentWire, PatchNode};
+
+pub struct KAKResynthesis{
+    wires: Vec<PersistentWire>
+}
+
+impl IterMatchedWires for KAKResynthesis {
+    fn matched_wires(&self) -> impl Iterator<Item = &PersistentWire> + '_ {
+        self.wires[0].iter().chain(self.wires[1].iter())
+    }
+}
+
+pub struct KAKFactory;
+
+struct SearchingWalker <'w> {
+    walker: Walker <'w>,                    // Initial state of walker.
+    frontier_wires: Vec<PersistentWire>,    // Wires to be expanded.
+    pattern: KAKResynthesis,                // Current pattern.
+    two_qubit_node: Option<PatchNode>,      // If a 2q gate has been found along one wire,
+                                            // but not yet the other, then it is stored here.
```

Comment:
Put the documentation for the fields on a separate line above the fields, starting with `///`. That way rust will automatically pick it up as docstrings (and include it in automatically generated documentation) :)

```suggestion
    /// Initial state of walker.
    walker: Walker <'w>,
    /// Wires to be expanded.
    frontier_wires: Vec<PersistentWire>,
    /// Current pattern.
    pattern: KAKResynthesis,
    /// If a 2q gate has been found along one wire,
    /// but not yet the other, then it is stored here.
    two_qubit_node: Option<PatchNode>,
```